transport/probe/src/db_proxy.rs

```diff
@@ -0,0 +1,41 @@
+use async_trait::async_trait;
+use hopr_db_api::{protocol::HoprDbProtocolOperations, resolver::HoprDbResolverOperations};
+
+use crate::traits::DbOperations;
+
+#[derive(Debug, Clone)]
+pub struct DbProxy<T>
+where
+    T: HoprDbResolverOperations + HoprDbProtocolOperations + std::fmt::Debug + Clone + Send + Sync + 'static,
```

Comment:
Generic bounds should be on `impls`, not on struct definitions. 

See e.g. https://stackoverflow.com/a/66369912

Reply:
I agree, and the truth is that this entire file needs to go away as a separate pipeline decorator rewritten to offer only the most basic surb injection and path resolution to isolate the upper layers. The implementation is here only so that I can do this at the current level. Once migrated away from probing, I can rewrite it to properly.

@coderabbitai please create in issue to extract the `db_proxy.rs` functionality from this file into a different decorator for the output pipeline and improve the code in terms of best practices. Assign the bug to me please.